### PR TITLE
[PKG-1992] rebuild for icu-73.1

### PR DESCRIPTION
**Pull request by rafaelmartins**
Depends on:
- https://github.com/AnacondaRecipes/icu-feedstock/pull/3
- https://github.com/AnacondaRecipes/libxml2-feedstock/pull/10

### CODE REVIEW

**Review by SC426 — APPROVED** [review]

**Review by JeanChristopheMorinPerso — COMMENTED** [review]
The linux-64 logs contains
```
WARNING (libgd): run-exports library package icu73/label/icu73::icu-73.1-h6a678d5_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

Looking at the linking commands, icu doesn't seems to be used... And looking at the code, I don't think ICU is used directly... Is it possible?

**Comment by rafaelmartins** [comment]
> The linux-64 logs contains
> 
> ```
> WARNING (libgd): run-exports library package icu73/label/icu73::icu-73.1-h6a678d5_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
> ```
> 
> Looking at the linking commands, icu doesn't seems to be used... And looking at the code, I don't think ICU is used directly... Is it possible?

It seems that we have several leftover dependencies in recipes. We need to think of a way to export this data to CI checks and make them fail something. I do not checked deeply the dependencies of pure rebuilds.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -39,22 +39,20 @@ requirements:
...
-    - libwebp-base
+    - libwebp-base 1.2
     - zlib 1.2.*
   run:
     - expat >=2.4.9,<3.0a0
     - fontconfig >=2.14.1,<3.0a0
     - freetype >=2.10.4,<3.0a0
-    - icu >=68.1,<69.0a0  # [s390x or aarch64 or (osx and arm64)]
-    - icu >=58.0,<59.0a0  # [not (s390x or aarch64 or (osx and arm64))]
+    - icu >=73.1,<74.0a0
```
Comment (boldorider4) [inline recipe/meta.yaml:55]: would icu `{{ pin_compatible('icu') }}` possibly avoid the hassle of doing this manually?

**Review by boldorider4 — APPROVED** [review]
Remember to remove abs.yaml

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:55]: Explicit is better than implicit for the runtime dependencies (and in general anyway).

**Comment by JeanChristopheMorinPerso** [comment]
>It seems that we have several leftover dependencies in recipes. We need to think of a way to export this data to CI checks and make them fail something. I do not checked deeply the dependencies of pure rebuilds.

It exists already. conda-build has `--error-overlinking` and `--error-overdepending`. Prefect by default uses these CLI flags https://github.com/anaconda-distribution/rocket-platform/blob/f3ad34ff01315506e0b51edc23e77f8b4a4bde76/cookbook/flows/config.yaml#L18-L21.

See https://github.com/anaconda-distribution/perseverance-skills/blob/main/sections/02_Package_building/06_Errors_and_troubleshooting/Overlinking_overdepending_dso_whitelist.md for an explanation of what overlinking and overdepending.

**Comment by rafaelmartins** [comment]
> > It seems that we have several leftover dependencies in recipes. We need to think of a way to export this data to CI checks and make them fail something. I do not checked deeply the dependencies of pure rebuilds.
> 
> It exists already. conda-build has `--error-overlinking` and `--error-overdepending`. Prefect by default uses these CLI flags https://github.com/anaconda-distribution/rocket-platform/blob/f3ad34ff01315506e0b51edc23e77f8b4a4bde76/cookbook/flows/config.yaml#L18-L21.
> 
> See https://github.com/anaconda-distribution/perseverance-skills/blob/main/sections/02_Package_building/06_Errors_and_troubleshooting/Overlinking_overdepending_dso_whitelist.md for an explanation of what overlinking and overdepending.

I know what overlinking is... What I want is something that raises a clear alert on this, like we have for the linter. in this case the builds are green, even if the warnings are part of the log. Ideally we would have something like a "post-build linter" that would collect these logs and fail if there are some conda warnings in the logs. Or maybe we could just make conda-build fail with warnings, like compilers do with `-Werror`. This `--error-overlinking` is not the same, as despite the name of the flag, it raises a non-fatal warning.

**Comment by JeanChristopheMorinPerso** [comment]
`--error-overlinking` and `--error-overdepending` should make the build fail.

For example, if I add `openssl` to harfbuzz and I build with `--error-overlinking --error-overdepending`, conda-build will fail with:
```
Command 'LINKER_OPTIMIZATION_HINT' not parsed!
   INFO: sysroot: '/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/' files: '['usr/share/man/mann/zero.n', 'usr/share/man/mann/yencode.n', 'usr/share/man/mann/yaml.n', 'usr/share/man/mann/xsxp.n']'
   INFO (harfbuzz,lib/libharfbuzz-subset.0.dylib): lib/libharfbuzz.0.dylib found in this package
   INFO (harfbuzz,lib/libharfbuzz-subset.0.dylib): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/usr/lib/libSystem.B.dylib found in $SYSROOT
   INFO (harfbuzz,lib/libharfbuzz.0.dylib): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/usr/lib/libSystem.B.dylib found in $SYSROOT
   INFO (harfbuzz,lib/libharfbuzz.0.dylib): Needed DSO lib/libfreetype.6.dylib found in pkgs/main::freetype-2.10.4-h1192e45_0
   INFO (harfbuzz,lib/libharfbuzz.0.dylib): Needed DSO lib/libgraphite2.3.dylib found in pkgs/main::graphite2-1.3.14-hc377ac9_1
   INFO (harfbuzz,lib/libharfbuzz.0.dylib): Needed DSO lib/libglib-2.0.0.dylib found in pkgs/main::glib-2.69.1-h514c7bf_2
   INFO (harfbuzz,lib/libharfbuzz.0.dylib): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation found in $SYSROOT
   INFO (harfbuzz,lib/libharfbuzz.0.dylib): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics found in $SYSROOT
   INFO (harfbuzz,lib/libharfbuzz.0.dylib): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/CoreText.framework/Versions/A/CoreText found in $SYSROOT
   INFO (harfbuzz,bin/hb-ot-shape-closure): lib/libharfbuzz.0.dylib found in this package
   INFO (harfbuzz,bin/hb-ot-shape-closure): Needed DSO lib/libglib-2.0.0.dylib found in pkgs/main::glib-2.69.1-h514c7bf_2
   INFO (harfbuzz,bin/hb-ot-shape-closure): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/usr/lib/libSystem.B.dylib found in $SYSROOT
   INFO (harfbuzz,lib/libharfbuzz-icu.0.dylib): lib/libharfbuzz.0.dylib found in this package
   INFO (harfbuzz,lib/libharfbuzz-icu.0.dylib): Needed DSO lib/libicuuc.73.dylib found in icu73/label/icu73::icu-73.1-h313beb8_0
   INFO (harfbuzz,lib/libharfbuzz-icu.0.dylib): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/usr/lib/libSystem.B.dylib found in $SYSROOT
   INFO (harfbuzz,bin/hb-view): lib/libharfbuzz.0.dylib found in this package
   INFO (harfbuzz,bin/hb-view): Needed DSO lib/libfreetype.6.dylib found in pkgs/main::freetype-2.10.4-h1192e45_0
   INFO (harfbuzz,bin/hb-view): Needed DSO lib/libcairo.2.dylib found in icu73/label/icu73::cairo-1.16.0-h27d1283_5
   INFO (harfbuzz,bin/hb-view): Needed DSO lib/libglib-2.0.0.dylib found in pkgs/main::glib-2.69.1-h514c7bf_2
   INFO (harfbuzz,bin/hb-view): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/usr/lib/libSystem.B.dylib found in $SYSROOT
   INFO (harfbuzz,bin/hb-shape): lib/libharfbuzz.0.dylib found in this package
   INFO (harfbuzz,bin/hb-shape): Needed DSO lib/libglib-2.0.0.dylib found in pkgs/main::glib-2.69.1-h514c7bf_2
   INFO (harfbuzz,bin/hb-shape): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/usr/lib/libSystem.B.dylib found in $SYSROOT
   INFO (harfbuzz,lib/libharfbuzz-gobject.0.dylib): lib/libharfbuzz.0.dylib found in this package
   INFO (harfbuzz,lib/libharfbuzz-gobject.0.dylib): Needed DSO lib/libglib-2.0.0.dylib found in pkgs/main::glib-2.69.1-h514c7bf_2
   INFO (harfbuzz,lib/libharfbuzz-gobject.0.dylib): Needed DSO lib/libgobject-2.0.0.dylib found in pkgs/main::glib-2.69.1-h514c7bf_2
   INFO (harfbuzz,lib/libharfbuzz-gobject.0.dylib): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/usr/lib/libSystem.B.dylib found in $SYSROOT
   INFO (harfbuzz,bin/hb-subset): lib/libharfbuzz.0.dylib found in this package
   INFO (harfbuzz,bin/hb-subset): lib/libharfbuzz-subset.0.dylib found in this package
   INFO (harfbuzz,bin/hb-subset): Needed DSO lib/libglib-2.0.0.dylib found in pkgs/main::glib-2.69.1-h514c7bf_2
   INFO (harfbuzz,bin/hb-subset): Needed DSO /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/usr/lib/libSystem.B.dylib found in $SYSROOT
  ERROR (harfbuzz): run-exports library package pkgs/main::openssl-1.1.1t-h1a28f6b_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
overdepending check failed 
['  ERROR (harfbuzz): run-exports library package pkgs/main::openssl-1.1.1t-h1a28f6b_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)']
(base) jcmorin-mac-anac:aggregate jcmorin$ 
```

**Comment by rafaelmartins** [comment]
> --error-overdepending

then this flag is what is missing. I'll add that to my test builds. thanks

**Pull request closed without merging**